crates/sdk/src/env.rs

```diff
@@ -33,10 +33,7 @@ fn expected_boolean<T>(e: u32) -> T {
 
 pub fn get_executor_identity() -> [u8; 32] {
     unsafe { sys::get_executor_identity(DATA_REGISTER) };
-    read_register(DATA_REGISTER)
-        .expect("Must have executor identity.")
-        .try_into()
-        .expect("Wrong executor identity length.")
+    read_register_sized(DATA_REGISTER).expect("Must have executor identity.")
```

Comment:
one thing you'll notice is we don't use the traditional panic because panics in WASM will abort the runtime with no information about what happened

so you have to call `panic_str` which will first send a contextual message to the host, after which the host terminates the execution

consider `unwrap_or_else`